core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/BaseAwsV4HttpSigner.java

```diff
@@ -106,10 +107,11 @@ default AsyncSignedRequest signAsync(AsyncSignRequest<? extends AwsCredentialsId
 
         Publisher<ByteBuffer> payload = processPayload(request.payload().orElse(null), futureV4RequestContext, properties);
 
-        return AsyncSignedRequest.builder()
-            .request(CompletableFutureUtils.joinLikeSync(futureV4RequestContext).getSignedRequest())
+        return futureV4RequestContext.thenApply(ctx -> AsyncSignedRequest.builder()
```

Comment:
Can we add a test case where we cancel the future captured from signAsync() ; in that case what happens to futureV4RequestContext ?